rules/generic/src/Rector/Assign/DimFetchAssignToMethodCallRector.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Generic\Rector\Assign;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ArrayDimFetch;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Generic\ValueObject\DimFetchAssignToMethodCall;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Generic\Tests\Rector\Assign\DimFetchAssignToMethodCallRector\DimFetchAssignToMethodCallRectorTest
+ */
+final class DimFetchAssignToMethodCallRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    public const DIM_FETCH_ASSIGN_TO_METHOD_CALL = 'dim_fetch_assign_to_method_call';
```

Comment:
what about just ASSIGN_TO_METHOD_CALL?

Reply:
Assign to method call is this:

```php
$this->someMethod() = 1000;
```

Assign to dim fetch is:

```php
$values[] = 1000;
```

New assing to dim fetch (case for this rule) is:

```php
$values[] = new SomeObject()
```

Reply:
I re-checked the rule now. Maybe `DimFetchAssignToMethodCall` would be better

Reply:
that's its name now :)

Reply:
Then it's verified :D